src/main/java/io/cdap/plugin/ariba/source/AribaBatchSource.java

```diff
@@ -84,6 +94,57 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
   @Override
   public void prepareRun(BatchSourceContext context) throws Exception {
     // TODO: https://cdap.atlassian.net/browse/PLUGIN-1172
+    FailureCollector collector = context.getFailureCollector();
+    pluginConfig.validatePluginParameters(collector); // validate when macros are already substituted
+    collector.getOrThrowException();
+
+    Schema outputSchema = context.getOutputSchema();
+    if (outputSchema == null) {
+      outputSchema = getOutputSchema();
+    }
+    if (outputSchema == null) {
+      throw new IllegalArgumentException(ResourceConstants.ERR_MACRO_INPUT.getMsgForKeyWithCode());
+    }
+    setJobForDataRead(context, outputSchema);
+    collector.getOrThrowException();
+  }
+
+  private Schema getOutputSchema() throws IOException, AribaException, InterruptedException {
+    String token = aribaServices.getAccessToken();
+    Schema schema = aribaServices.buildOutputSchema(token);
+    LOG.trace("Initiating Metadata Call To Ariba");
+    return schema;
+  }
+
+  private void setJobForDataRead(BatchSourceContext context, Schema outputSchema) throws IOException {
+    Job job = JobUtils.createInstance();
+    Configuration jobConfiguration = job.getConfiguration();
+    Gson gson = new GsonBuilder().create();
```

Comment:
create a static instance similar to what you have done in the input format
```
private static final Gson GSON = new Gson();
```

Reply:
Fixed